src/50_Samples_%26_Templates/Product.html

```diff
@@ -198,6 +245,24 @@
     <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
     </head>
     <body>
+    <amp-state id="allColors">
```

Comment:
This data structure feels a bit weird. It's ultimately tabular data, but stored as a associative array, which sometimes has key names ('availableSizes') and sometimes drops key names ('color'). Also, it has a very specific name: `allColors`. (In a production-ready version, since this will all need to be machine generated, it would probably end up something generic like `product` or `price`.)

I'm wondering if it would make things simpler if each dimension (color, size, price, etc.) were mapped to successive levels of the associative array:

```javascript
{
  'green': {
    'S': null,
    'M': '$1.99',
    'L': '$1.99'
  },
  'golden': {
    'S': '$9.99',
    ...
  }   
}
```

This can then be indexed as e.g. `price[color][size]`.